integration-tests/plugins/__tests__/cart/store/add-promotions-to-cart.spec.ts

```diff
@@ -0,0 +1,309 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { ICartModuleService, IPromotionModuleService } from "@medusajs/types"
+import { PromotionType } from "@medusajs/utils"
+import path from "path"
+import { startBootstrapApp } from "../../../../environment-helpers/bootstrap-app"
+import { useApi } from "../../../../environment-helpers/use-api"
+import { getContainer } from "../../../../environment-helpers/use-container"
+import { initDb, useDb } from "../../../../environment-helpers/use-db"
+import adminSeeder from "../../../../helpers/admin-seeder"
+
+jest.setTimeout(50000)
+
+const env = { MEDUSA_FF_MEDUSA_V2: true }
+
+describe("Store Carts API: Add promotions to cart", () => {
+  let dbConnection
+  let appContainer
+  let shutdownServer
+  let cartModuleService: ICartModuleService
+  let promotionModuleService: IPromotionModuleService
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", "..", ".."))
+    dbConnection = await initDb({ cwd, env } as any)
+    shutdownServer = await startBootstrapApp({ cwd, env })
+    appContainer = getContainer()
+    cartModuleService = appContainer.resolve(ModuleRegistrationName.CART)
+    promotionModuleService = appContainer.resolve(
+      ModuleRegistrationName.PROMOTION
+    )
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+    await shutdownServer()
+  })
+
+  beforeEach(async () => {
+    await adminSeeder(dbConnection)
+  })
+
+  afterEach(async () => {
+    const db = useDb()
+    await db.teardown()
+  })
+
+  describe("POST /store/carts/:id/promotions", () => {
+    it("should add line item adjustments to a cart based on promotions", async () => {
+      const appliedPromotion = await promotionModuleService.create({
+        code: "PROMOTION_APPLIED",
+        type: PromotionType.STANDARD,
+        application_method: {
+          type: "fixed",
+          target_type: "items",
+          allocation: "each",
+          value: "300",
+          apply_to_quantity: 1,
+          max_quantity: 1,
+          target_rules: [
+            {
+              attribute: "product_id",
+              operator: "eq",
+              values: "prod_tshirt",
+            },
+          ],
+        },
+      })
+
+      const createdPromotion = await promotionModuleService.create({
+        code: "PROMOTION_TEST",
+        type: PromotionType.STANDARD,
+        application_method: {
+          type: "fixed",
+          target_type: "items",
+          allocation: "across",
+          value: "1000",
+          apply_to_quantity: 1,
+          target_rules: [
+            {
+              attribute: "product_id",
+              operator: "eq",
+              values: "prod_mat",
+            },
+          ],
+        },
+      })
+
+      const cart = await cartModuleService.create({
+        currency_code: "usd",
+        items: [
```

Comment:
Maybe we can create some fixtures that you can reuse, esp. in tests where the creation is just a matter of preparing for the actual test.